### PR TITLE
remove flatcc dependency

### PR DESCRIPTION
**Pull request by MichaelBrim**
### Description

This removes our use of flatcc for serializing the array of read requests during the mread rpc. The serialization provides no benefit since the data is only shared between client and server on the same host. We now just pack the array of requests into the mercury bulk buffer. The documentation and support scripts have been updated accordingly.

Also:
- fixed a few unused variable warnings
- clarified code comments that talked about "delegators" to explicitly use "server" or "request manager thread / reqmgr" as appropriate

### Motivation and Context

See #550 

### How Has This Been Tested?

Used `read` and `writeread` examples in `listio` mode on Summitdev. Note that listio mode in the examples are the only tests/examples that currently exercise the mread path.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Testing (addition of new tests or update to current tests)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyFS code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

**Comment by MichaelBrim** [comment]
@CamStan Just a reminder that when this is merged, we will need to update our spack recipe too.

File: client/src/unifyfs.c [hunk 1]
```diff
@@ -1059,39 +1058,28 @@ int unifyfs_gfid_read_reqs(read_req_t* in_reqs, int in_count)
...
-        flatcc_builder_t builder;
-        flatcc_builder_init(&builder);
-
-        /* create request vector */
-        unifyfs_Extent_vec_start(&builder);
-
-        /* fill in values for each request entry */
+        /* got multiple read requests */
+        size_t size = (size_t)count * sizeof(unifyfs_extent_t);
+        void* buffer = malloc(size);
```
Comment (CamStan) [inline client/src/unifyfs.c:1063]: You removed the `free(buffer);` below on line 1094. Still need it for this?

Comment (MichaelBrim) [reply client/src/unifyfs.c:1063]: Good catch - it's still needed.

Comment (MichaelBrim) [reply client/src/unifyfs.c:1063]: for flatcc.m4, just missed it when removing files

Comment (MichaelBrim) [reply client/src/unifyfs.c:1063]: I just pushed updates for these two oversights.

**Comment by CamStan** [comment]
Related PR for Spack package update: https://github.com/spack/spack/pull/17913

**Comment by adammoody** [comment]
Hooray!!  Nice clean up @MichaelBrim !  And thanks for the review @CamStan

**Pull request merged**